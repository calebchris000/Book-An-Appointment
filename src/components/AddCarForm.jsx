import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { addCar } from "../redux/signup/signupSlice";
const AddCarForm = () => {
  const dispatch = useDispatch();
  const {isLoading, error, successMessage }= useSelector((store) => store.signup);
  const [name, setName] = useState("");
  const [image, setImage] = useState("");
  const [description, setDescription] = useState("");
  const [model, setModel] = useState("");
  const [engine, setEngine] = useState("");
  const [price, setPrice] = useState("");
  const [mileage, setMileage] = useState("");
  const handleSubmit = (e) => {
    e.preventDefault();
    const userData = {
      name,
      image,
      description,
      model,
      engine,
      price,
      mileage,
    };
    const requestOptions = {
      sign_in: true,
      endPoints: "api/products",
      method: {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(userData)
    }};
    dispatch(addCar(requestOptions));
  };
  return (
    <div className="container-md">
      <h2>Add Car</h2>
      {successMessage && <div className="success-message">{successMessage}</div>}
      {error && <div className="error-message">{error}</div>}
      <form onSubmit={handleSubmit}>
        <div className="mb-3">
          <label>Name:</label>
          <input type="text" value={name} onChange={(e) => setName(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Image:</label>
          <input type="text" value={image} onChange={(e) => setImage(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Description:</label>
          <input type="text" value={description} onChange={(e) => setDescription(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Model:</label>
          <input type="text" value={model} onChange={(e) => setModel(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Engine:</label>
          <input type="text" value={engine} onChange={(e) => setEngine(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Price:</label>
          <input type="integer" value={price} onChange={(e) => setPrice(e.target.value)} required />
        </div>
        <div className="mb-3">
          <label>Mileage:</label>
          <input type="integer" value={mileage} onChange={(e) => setMileage(e.target.value)} required />
        </div>
        <button className = "btn btn-primary" type="submit" disabled={isLoading}>
          {isLoading ? "Loading..." : "Add Car"}
        </button>
      </form>
    </div>
  );
};
export default AddCarForm;