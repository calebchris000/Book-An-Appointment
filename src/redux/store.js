import { configureStore } from "@reduxjs/toolkit";
import productDetailsReducer from "./productDetails/productDetailsSlice";
import productDescriptionReducer from "./productDescription/productDescriptionSlice";
import thunk from "redux-thunk";
import deleteCarReducer from './deleteCarSlice';

const store = configureStore({
  reducer: {
    productDetails: productDetailsReducer,
    productDescription: productDescriptionReducer,
  },
  middleware: [thunk],

  reducer: {
    deleteCar: deleteCarReducer,
  },
});

export default store;
