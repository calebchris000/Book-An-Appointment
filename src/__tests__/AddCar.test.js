import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import AddCarForm from '../components/Addcars/AddCars';


it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <AddCarForm />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});