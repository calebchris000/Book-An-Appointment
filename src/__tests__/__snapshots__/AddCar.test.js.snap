// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="add-car-form"
>
  <img
    className="reservation-image-overlay"
    src="https://i.ibb.co/pK2TsyG/486-4862054-super-car-png.png"
  />
  <h2
    className="add-car-title"
  >
    ADD A CAR
  </h2>
  <form
    className="add-car-form-component"
    onSubmit={[Function]}
  >
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Name:
      </label>
      <input
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Image:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Description:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Model:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Engine:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Price:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="integer"
        value=""
      />
    </div>
    <div
      className=""
    >
      <label
        className="add-car-label"
      >
        Mileage:
      </label>
      <input
        className="add-car-input"
        onChange={[Function]}
        required={true}
        type="integer"
        value=""
      />
    </div>
    <button
      className="submit-add-car"
      disabled={false}
      type="submit"
    >
      Add Car
    </button>
  </form>
</div>
`;
